aisdc/postprocessing/generate_report.py

```diff
@@ -0,0 +1,283 @@
+import json
+import os
+import pprint
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+
+class AnalysisModule:
+    """
+    Wrapper module for metrics analysis modules
+    """
+
+    def process_dict(self):
+        """
+        Function that produces a risk summary output based on analysis in this module
+        """
+        raise NotImplementedError()
+
+    def __str__(self):
+        raise NotImplementedError()
+
+
+class FinalRecommendationModule(AnalysisModule):
+    """
+    Module that generates the first layer of a recommendation report
+    """
+
+    def __init__(self, report: dict):
+        self.P_VAL_THRESH = 0.1
+        self.MEAN_AUC_THRESH = 0.65
+
+        self.SVM_WEIGHTING_SCORE = 5
+        self.MIN_SAMPLES_LEAF_SCORE = 3
+        self.STATISTICALLY_SIGNIFICANT_SCORE = 2
+        self.MEAN_AUC_SCORE = 4
+
+        self.report = report
+
+        self.scores = []
+        self.reasons = []
+
+    def _is_svm(self, svm_weighing_score):
```

Comment:
So, something to check (not now). Jim et al made a SAFE SVC that wraps a SVC trained with differentially private training. So just need to ensure that the self.report["model"] field isn't SVC in that (i can see how python inheritance might mean that it is).